### PR TITLE
Bugfix: android exception manager crashes

### PR DESCRIPTION
**Pull request by estebanmino**
**Description**

Fixes 

- https://www.fabric.io/metamask/android/apps/io.metamask/issues/5c3f843bf8b88c29637fb4f9/sessions/5DC5935C0348000167AE5D826E436941_DNE_0_v2?build=114539895

- https://www.fabric.io/metamask/android/apps/io.metamask/issues/5c3f843bf8b88c29637fb4f9/sessions/5DC5BE020129000173471229A6054B91_DNE_0_v2?build=114539895

- Adds a nee method `safeToChecksumAddress` for cleaner code in places we need to check for undefined values trying to be checksummed.

- Uses https://github.com/MetaMask/gaba/pull/178 to update `selectedAddress`, that way we're sure that we're handling a checksummed address.

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

File: app/components/UI/AssetOverview/index.js [hunk 1]
```diff
@@ -148,7 +148,7 @@ class AssetOverview extends PureComponent {
 			currentCurrency
 		} = this.props;
 		let mainBalance, secondaryBalance;
-		const itemAddress = (asset.address && toChecksumAddress(asset.address)) || undefined;
+		const itemAddress = safeToChecksumAddress(asset.address) || undefined;
```
Comment (brunobar79) [inline app/components/UI/AssetOverview/index.js]: you don't need the || here

File: app/components/UI/Tokens/index.js [hunk 2]
```diff
@@ -135,7 +135,7 @@ class Tokens extends PureComponent {
 
 	renderItem = asset => {
 		const { conversionRate, currentCurrency, tokenBalances, tokenExchangeRates, primaryCurrency } = this.props;
-		const itemAddress = (asset.address && toChecksumAddress(asset.address)) || undefined;
+		const itemAddress = safeToChecksumAddress(asset.address) || undefined;
```
Comment (brunobar79) [inline app/components/UI/Tokens/index.js]: No need for ||

File: app/util/transactions.js [hunk 3]
```diff
@@ -244,8 +245,8 @@ export async function getActionKey(tx, selectedAddress, ticker, paymentChannelTr
 	const actionKey = await getTransactionActionKey(tx);
 	if (actionKey === SEND_ETHER_ACTION_KEY) {
 		ticker = paymentChannelTransaction ? strings('unit.dai') : ticker;
-		const incoming = toChecksumAddress(tx.transaction.to) === toChecksumAddress(selectedAddress);
-		const selfSent = incoming && toChecksumAddress(tx.transaction.from) === toChecksumAddress(selectedAddress);
+		const incoming = safeToChecksumAddress(tx.transaction.to) === selectedAddress;
+		const selfSent = incoming && safeToChecksumAddress(tx.transaction.from) === selectedAddress;
```
Comment (brunobar79) [inline app/util/transactions.js]: I don't think you need the `incoming &&` anymore

**Review by brunobar79 — APPROVED** [review]
Minor stuff but LGTM

**Comment by ibrahimtaveras00** [comment]
Looks good, QA Passed 👍

**Review by ibrahimtaveras00 — APPROVED** [review]

File: app/components/UI/AssetOverview/index.js (hunk 1 above)
Comment (estebanmino) [reply app/components/UI/AssetOverview/index.js]: 🤦‍♂

File: app/util/transactions.js (hunk 3 above)
Comment (estebanmino) [reply app/util/transactions.js]: is comparing `to` against `from` to know if `selfSent`

**Pull request merged**